### PR TITLE
TASK: Improvements to the UserInitials view helper

### PR DESCRIPTION
**Pull request by aertmann**
Improves the UserInitials view helper a bit by removing the odd dot and the end of "fullFirstName" and trims the result. Additionally strips special characters from "initials".

### CODE REVIEW

**Comment by aertmann** [comment]
image
See the broken initials and the whitespace before the dot in the name

**Comment by bwaidelich** [comment]
Right - looking at it again, they seem to have put "{aalborgstift.dk) Christian Roar Pedersen" as `firstName` and `lastName` is empty?!

**Comment by aertmann** [comment]
Ah good catch @kdambekalns! What to do in such a case? Maybe remove the space and dot if no last name exists?

**Comment by aertmann** [comment]
okay fixed it @kdambekalns @bwaidelich

File: TYPO3.Neos/Classes/TYPO3/Neos/ViewHelpers/Backend/UserInitialsViewHelper.php [hunk 1]
```diff
@@ -74,9 +75,9 @@ public function render($format = 'initials')
 
         switch ($format) {
             case 'initials':
-                return mb_substr($requestedUser->getName()->getFirstName(), 0, 1) . mb_substr($requestedUser->getName()->getLastName(), 0, 1);
+                return mb_substr(preg_replace('/[^[:alnum:][:space:]]/u', '', $requestedUser->getName()->getFirstName()), 0, 1) . mb_substr($requestedUser->getName()->getLastName(), 0, 1);
```
Comment (bwaidelich) [inline TYPO3.Neos/Classes/TYPO3/Neos/ViewHelpers/Backend/UserInitialsViewHelper.php]: I didn't realize this the first time around, but now I wonder: Why do you remove special characters from the first name but not from the last name?

Comment (aertmann) [reply TYPO3.Neos/Classes/TYPO3/Neos/ViewHelpers/Backend/UserInitialsViewHelper.php]: good point, should be done to both.. fixed

**Comment by aertmann** [comment]
thanks!

**Pull request merged**